src/Generator/RequestGenerator.php

```diff
@@ -289,4 +254,77 @@ protected function generateGetBody(?Field $body): ClassMethod
             ->addStmt($this->builder->return($this->builder->val(null)))
             ->getNode();
     }
+
+    private function generateGetHeadersMethod(Request $request, array $fields): ClassMethod
+    {
+        $headers = [];
+        if ($request->getContentType() !== '') {
+            $headers['Content-Type'] = $this->builder->val($request->getContentType());
+        }
+        $returnVal  = $this->builder->array($headers);
+        $fieldsArr  = [];
+        $returnType = 'array';
+        foreach ($fields as $field) {
+            $fieldsArr[$field->getName()] = $this->builder->localPropertyFetch($field->getPhpVariableName());
+        }
+
+        if (!empty($fieldsArr)) {
+            $returnVal = $this->builder->funcCall(
+                'array_merge',
+                [$returnVal, $this->generateParametersFromFields($fieldsArr)]
+            );
+        }
+
+        return $this->builder
+            ->method('getHeaders')
+            ->makePublic()
+            ->addStmt($this->builder->return($returnVal))
+            ->setReturnType($returnType)
+            ->composeDocBlock([], $returnType)
+            ->getNode();
+    }
+
+    private function generateParametersFromFields(array $fields): FuncCall
+    {
+        $filterCallbackBody = $this->builder->return(
+            $this->builder->notEquals($this->builder->val(null), $this->builder->var('value'))
+        );
+
+        $filterCallback = $this->builder->closure(
+            [$filterCallbackBody],
+            [$this->builder->param('value')->getNode()]
+        );
+
+        $filter = $this->builder->funcCall(
+            'array_filter',
+            [$this->builder->array($fields), $filterCallback]
+        );
+
+        $this->addImport(CopiedNamespace::getImport($this->baseNamespace, SerializableInterface::class));
+        $closureVariable = $this->builder->var('value');
+        $closureBody     = $this->builder->return(
+            $this->builder->ternary(
+                $this->builder->instanceOf(
+                    $closureVariable,
+                    $this->builder->className('SerializableRequestBody')
```

Comment:
Please replace by `$this->builder->className('SerializableInterface')`.

Reply:
Done